mobile/android/android-emulator-services.ts

```diff
@@ -120,29 +147,113 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 		return timeout * 1000;
 	}
 
+
+	private getRunningEmulatorId(image: string): IFuture<string> {
+		return ((): string => {
+			var runningEmulators = this.getRunningEmulators().wait();
+			if(runningEmulators.length === 0) {
+				return "";
+			}
+
+			// if we get here, there's at least one running emulator
+			var emulatorId: string;
+
+			_(runningEmulators).each(emulator => {
+				var emulatorName = options.geny ? this.getNameFromGenymotionEmulatorId(emulator).wait() : this.getNameFromNativeEmulatorId(emulator).wait();
+				if(emulatorName === image) {
```

Comment:
I would move the function choice outside of the for each cycle to shorten this line a bit.